server/graphql/v1/mutations/expenses.js

```diff
@@ -703,67 +705,93 @@ export async function payExpense(req, args) {
+        if (args.forceManual) {
+          feesInHostCurrency.paymentProcessorFeeInHostCurrency = 0;
+          await createTransactions(host, expense, feesInHostCurrency);
+        } else {
+          await payExpenseWithTransferwise(host, payoutMethod, expense, feesInHostCurrency, remoteUser);
+          await expense.setProcessing(remoteUser.id);
+          // Early return, we'll only mark as Paid when the transaction completes.
+          return expense;
+        }
+      } else if (payoutMethodType === PayoutMethodTypes.ACCOUNT_BALANCE) {
+        const payee = expense.fromCollective;
+        const payeeHost = await payee.getHostCollective();
+        if (!payeeHost) {
+          throw new Error('The payee needs to have an Host to able to be paid on its Open Collective balance.');
+        }
+        if (host.id !== payeeHost.id) {
+          throw new Error(
+            'The payee needs to be on the same Host than the payer to be paid on its Open Collective balance.',
+          );
+        }
         await createTransactions(host, expense, feesInHostCurrency);
-      } else if (paypalPaymentMethod) {
-        await payExpenseWithPayPal(remoteUser, expense, host, paypalPaymentMethod, paypalEmail, feesInHostCurrency);
-      } else {
-        throw new Error('No Paypal account linked, please reconnect Paypal or pay manually');
-      }
-    } else if (payoutMethodType === PayoutMethodTypes.BANK_ACCOUNT) {
-      if (args.forceManual) {
-        feesInHostCurrency.paymentProcessorFeeInHostCurrency = 0;
+      } else if (expense.legacyPayoutMethod === 'manual' || expense.legacyPayoutMethod === 'other') {
+        // note: we need to check for manual and other for legacy reasons
         await createTransactions(host, expense, feesInHostCurrency);
-      } else {
-        await payExpenseWithTransferwise(host, payoutMethod, expense, feesInHostCurrency, remoteUser);
-        await expense.setProcessing(remoteUser.id);
-        // Early return, we'll only mark as Paid when the transaction completes.
-        return expense;
-      }
-    } else if (payoutMethodType === PayoutMethodTypes.ACCOUNT_BALANCE) {
-      const payee = expense.fromCollective;
-      const payeeHost = await payee.getHostCollective();
-      if (!payeeHost) {
-        throw new Error('The payee needs to have an Host to able to be paid on its Open Collective balance.');
       }
-      if (host.id !== payeeHost.id) {
-        throw new Error(
-          'The payee needs to be on the same Host than the payer to be paid on its Open Collective balance.',
-        );
+    } catch (error) {
+      // if the payment doesn't go through we want to reset the cache amount
+      if (currentRollingPayoutLimit) {
+        const { cacheKey, cacheValue } = currentRollingPayoutLimit;
+        const resetCacheValue = cacheValue === 0 ? cacheValue : cacheValue - expense.amount;
+        cache.set(cacheKey, resetCacheValue, CACHE_VALIDITY);
```

Comment:
Maybe we should simplify the ternary with an explicit conditional:
```suggestion
        if (cacheValue !== 0) {
          cache.set(cacheKey, cacheValue - expense.amount, CACHE_VALIDITY);
        }
```